### ISSUE TITLE
Setting seed for SimCLR transforms

### ISSUE THREAD

**Issue by amitness** [PyTorchLightning/pytorch-lightning-bolts#156]
Hi @williamFalcon,

Currently, the SimCLR implementation uses a set of transformations that are stochastic. Is there any way to set the seed so that subsequent experiments are reproducible? 

I was using `overfit_batches=1` and `batch_size=2` to just make sure SimCLR overfits correctly. The batch has the same 2 images loaded, but on each epoch, the transformations applied is different.

**Comment by amitness** [comment]
@williamFalcon 
Also, here is a minimal code to reproduce it: The model doesn't overfit at all even on a single batch with just 2 images.

```
from pl_bolts.models.self_supervised import SimCLR
import pytorch_lightning as pl

pl.seed_everything(42)

model = SimCLR(data_dir='/tmp', batch_size=2)

trainer = pl.Trainer(gpus=1, 
                     overfit_batches=1,
                     deterministic=True)
trainer.fit(model)
```
